### PR TITLE
Topic recognition/DCGAN in OASIS

### PR DESCRIPTION
**Pull request by xD-Jie**
From COMP3710 student, Jie Deng 45607316

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Sufficiently detailed commit messages
- Code valid for Hard Difficulty 
- Results are good

Other Feedback
- Should space out your commits. e.g. each commit should contain a self-contained chunk of work.\
- Please group the `.jpg/gif` files and put them in a sub-directory
- Good code structure

**Comment by rexsw** [comment]
Code looks good for hard difficulty

Good use of tensorflow

Comments like
     # Set loss function
      cross_entropy = tf.keras.losses.BinaryCrossentropy(from_logits=True)
Do no improve code readability and should be left out
Overall the code could use more meaningful comments

File: recognition/OASIS_DCGAN/dcgan.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import tensorflow as tf
```
Comment (shakes76) [inline recognition/OASIS_DCGAN/dcgan.py:1]: Code header block should be added for all scripts/source files

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF NOT submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Submit PDF
No Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.
